### ISSUE TITLE
paste preserving formatting

### ISSUE THREAD

**Issue by kodbeyi** [xournalpp/xournalpp#5750]
### Is Your Feature Request Related to a Problem?

When I paste the colored text I copied into the text box in the application, the text turns black. 
I want the text to stick in color.

### Feature Description

I want the text I copied in color to be pasted into the text box in the application, preserving its color.

**Comment by kodbeyi** [comment]
@rolandlo 
I'm following the nightly version, you didn't add this feature.

**Comment by rolandlo** [comment]
> @rolandlo I'm following the nightly version, you didn't add this feature.

No, I didn't. Did I claim so? The pasted text takes the current color of the text tool in Xournal++.
By the way, from which application are you copying?

**Comment by kodbeyi** [comment]
@rolandlo 
I thought you didn't see it because you didn't reply. Sorry.

https://github.com/user-attachments/assets/0075fc5c-032e-4863-a8e8-734ce39b886c

**Comment by rolandlo** [comment]
By the file format text elements in Xournal++ can only have one color. So there is no easy fix for this.
>
```
<text font="..." size="..." x="..." y="..." color="...">... text ...</text>
```
>The font attribute contains the font name, for example "Serif Bold Italic"; if the font is not available, another font will be substituted. The size attribute specifies the font size in points. The x and y attributes specify the coordinates of the top-left corner of the text box in page coordinates (measured in points from the top-left corner of the page). Finally, the color attribute contains either the name of a standard color or a hexadecimal RGBA value (see above).

**Comment by kodbeyi** [comment]
@rolandlo 
Wouldn't it be possible if you added this application as a button and when you clicked it, it would change the color of my texts according to the rules?
[color.zip](https://github.com/user-attachments/files/16635799/color.zip)

**Comment by rolandlo** [comment]
No it's not possible, as each text element can only have one color. You can't have different colors for different characters in Xournal++.

**Comment by kodbeyi** [comment]
@rolandlo 

I want to ask one more thing. Can't you make it possible to paste the text I copied as an image? I think this feature was available in the Microsoft Word application.

**Comment by rolandlo** [comment]
That depends on how the application puts the content you copy into the clipboard. If there is image data, Xournal++ could use it (with an option like special paste).
